### PR TITLE
refactor(generator): split DescriptorPoolFixture

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #7664

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10676)

### CODE REVIEW

File: generator/testing/descriptor_pool_fixture.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+   * Adds a new proto file and "compiles" it.
+   *
+   * Typically used to set up the conditions of a test, as in:
+   *
+   * @code
+   * class MyTest : public DescriptorPoolFixture {};
+   *
+   * TEST_F(MyTest, Name) {
+   *   ASSERT_TRUE(AddProtoFile("foo.proto", contents));
+   * }
```
Comment (dbolduc) [inline generator/testing/descriptor_pool_fixture.h:64]: nit: `@endcode` ?

File: generator/testing/descriptor_pool_fixture.h [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace generator_testing {
+
+/**
+ * Implements a fixture for test using a `google::protobuf::DescriptorPool`.
+ *
+ * Some test need a properly initialized descriptor pool, with:
```
Comment (dbolduc) [inline generator/testing/descriptor_pool_fixture.h:31]: s/test/tests/

File: generator/testing/descriptor_pool_fixture.h [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ * - the basic protobuf types and options already available
+ * - connected to error collectors so the test fails with meaningful errors
```
Comment (dbolduc) [inline generator/testing/descriptor_pool_fixture.h:33]: s/connected to//

File: generator/testing/descriptor_pool_fixture.h [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+#include <google/protobuf/descriptor.h>
+#include <google/protobuf/descriptor_database.h>
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace generator_testing {
+
+/**
+ * Implements a fixture for test using a `google::protobuf::DescriptorPool`.
```
Comment (dbolduc) [inline generator/testing/descriptor_pool_fixture.h:29]: optional: s/test using/testing with/ ?

**Review by dbolduc — APPROVED** [review]

File: generator/testing/descriptor_pool_fixture.h (hunk 1 above)
Comment (coryan) [reply generator/testing/descriptor_pool_fixture.h:64]: Thanks, fixed.

File: generator/testing/descriptor_pool_fixture.h (hunk 4 above)
Comment (coryan) [reply generator/testing/descriptor_pool_fixture.h]: Fixed

File: generator/testing/descriptor_pool_fixture.h (hunk 3 above)
Comment (coryan) [reply generator/testing/descriptor_pool_fixture.h]: Fixed

File: generator/testing/descriptor_pool_fixture.h (hunk 2 above)
Comment (coryan) [reply generator/testing/descriptor_pool_fixture.h]: Fixed

**Pull request merged**